File: DM&ADS/DP/L.java
package AlgoLab3;

import javafx.util.Pair;

import java.io.*;
import java.lang.reflect.Array;
import java.util.*;

public class L {
    static int n;
    static int w;

    void solve(PrintWriter out) throws IOException {
        FastScanner in = new FastScanner("skyscraper.in");
        n=in.nextInt();
        w=in.nextInt();
        ArrayList<Integer> d=new ArrayList<>();
        ArrayList<Integer> c=new ArrayList<>();
        for (int i = 0; i <n ; i++) {
            c.add(in.nextInt());
            d.add(1+i);
        }
        Collections.reverse(c);
        Collections.reverse(d);
        System.out.print(Integer.MAX_VALUE);





    }


    StreamTokenizer in;
    PrintWriter out;

    void run() throws IOException {
        out = new PrintWriter("skyscraper.out");
        solve(out);
        out.flush();
    }

    public static void main(String[] args) throws IOException {
        new L().run();

    }

    class FastScanner {
        BufferedReader reader;
        StringTokenizer tokenizer;

        public FastScanner(String fileName) throws IOException {
            reader = new BufferedReader(new FileReader(fileName));
        }

        public String next() throws IOException {
            while (tokenizer == null || !tokenizer.hasMoreTokens()) {
                String line = reader.readLine();
                if (line == null || line.equals("")) {
                    return "";
                }
                tokenizer = new StringTokenizer(line);
            }
            return tokenizer.nextToken();
        }

        public int nextInt() throws IOException {
            return Integer.parseInt(next());
        }

        public double nextDouble() throws IOException {
            return Double.parseDouble(next());
        }
    }

}

File: ML/cf/L.kt
import javafx.scene.text.FontWeight
import java.io.*
import java.util.*
import kotlin.collections.ArrayList
import kotlin.math.pow
import kotlin.math.sqrt
import kotlin.random.Random

object L {
    val fileName = ""

    //================================================================================


    fun run() {
        val N = IO.nextInt()
        val objs = ArrayList<Pair<Int,Int>>()
        repeat(N) {
            objs.add(Pair(IO.nextInt(), IO.nextInt()))
        }
        val av1 = objs.map { it.first }.average()
        val av2 = objs.map { it.second }.average()
        var cov = 0.0
        var sum1 = 0.0
        var sum2 = 0.0
        for (obj in objs) {
            cov += (obj.first.toDouble() - av1) * (obj.second.toDouble() - av2)
            sum1 += (obj.first.toDouble() - av1) * (obj.first.toDouble() - av1)
            sum2 += (obj.second.toDouble() - av2) * (obj.second.toDouble() - av2)
        }
        val denom = sqrt(sum1*sum2)
        if (denom == 0.0) {
            IO.println(0.0)
        } else {
            IO.println("%.10f".format(cov / denom))
        }

    }
    //==================================================================================

    @JvmStatic
    fun main(args: Array<String>) {
        this.run()
        IO.writer.flush()
        IO.reader.close()

    }

    object IO {
        val reader: BufferedReader
        val writer: PrintWriter
        private var tokenizer: StringTokenizer? = null

        init {
            if (fileName.isEmpty()) {
                reader = BufferedReader(InputStreamReader(System.`in`))
                writer = PrintWriter(System.out)
            } else {
                reader = BufferedReader(FileReader("$fileName.in"))
                writer = PrintWriter(FileWriter("$fileName.out"))
            }
        }

        fun next(): String {
            while (tokenizer == null || !tokenizer!!.hasMoreTokens()) {
                val line = reader.readLine() ?: throw NullPointerException()
                if (line == "") {
                    return ""
                }
                tokenizer = StringTokenizer(line)
            }
            return tokenizer!!.nextToken()
        }

        fun nextLine(): String {
            return reader.readLine()
        }

        fun nextInt(): Int {
            return next().toInt()
        }

        fun nextDouble(): Double {
            return next().toDouble()
        }

        fun nextLong(): Long {
            return next().toLong()
        }

        fun <T> print(obj: T) {
            writer.print(obj)
        }

        fun <T> println(obj: T) {
            writer.println(obj)
        }
    }
}